src/Sprites/spriteMap.ts

```diff
@@ -0,0 +1,659 @@
+import { Nullable } from "../types";
+import { Vector2, Vector3 } from "../Maths/math.vector";
+import { Texture } from "../Materials/Textures/texture";
+import { RawTexture } from "../Materials/Textures/rawTexture";
+import { ShaderMaterial } from "../Materials/shaderMaterial";
+import { Mesh } from "../Meshes/mesh";
+import { PickingInfo } from "../Collisions/pickingInfo";
+
+import "../Meshes/Builders/planeBuilder";
+
+/**
+ * Defines the basic options interface of a SpriteMap
+ */
+export interface ISpriteMapOptions{
+
+    /**
+	 * Vector2 of the number of cells in the grid.
+	 */
+    stageSize?: Vector2;
+
+    /**
+	 * Vector2 of the size of the output plane in World Units.
+	 */
+    outputSize?: Vector2;
+
+    /**
+	 * Vector3 of the position of the output plane in World Units.
+	 */
+    outputPosition?: Vector3;
+
+    //TODO ROTATION
+
+    /**
+	 * number of layers that the system will reserve in resources.
+	 */
+    layerCount?: number;
+
+    /**
+	 * number of max animation frames a single cell will reserve in resources.
+	 */
+    maxAnimationFrames?: number;
+
+    /**
+	 * number cell index of the base tile when the system compiles.
+	 */
+    baseTile?: number;
+
+    /**
+	* boolean flip the sprite after its been repositioned by the framing data.
+	*/
+    flipU?: boolean;
+
+    /**
+	 * Vector3 scalar of the global RGB values of the SpriteMap.
+	 */
+    colorMultiply?: Vector3;
+
+}
+
+/**
+ * Defines the basic options interface of a Sprite Frame Source Size.
+ */
+export interface ISpriteJSONSpriteSourceSize{
+    /**
+	 * number of the original width of the Frame
+	 */
+    w : number;
+
+    /**
+     * number of the original height of the Frame
+     */
+    h : number;
+}
+
+/**
+ * Defines the basic options interface of a Sprite Frame Data.
+ */
+export interface ISpriteJSONSpriteFrameData{
+    /**
+	 * number of the x offset of the Frame
+	 */
+    x : number;
+
+    /**
+	 * number of the y offset of the Frame
+	 */
+    y : number;
+
+    /**
+	 * number of the width of the Frame
+	 */
+    w : number;
+
+    /**
+     * number of the height of the Frame
+     */
+    h : number;
+}
+
+/**
+ * Defines the basic options interface of a JSON Sprite.
+ */
+export interface ISpriteJSONSprite{
+    /**
+	 * string name of the Frame
+	 */
+    filename : string;
+
+    /**
+	 * ISpriteJSONSpriteFrame basic object of the frame data
+	 */
+    frame : ISpriteJSONSpriteFrameData;
+
+    /**
+    * boolean to flag is the frame was rotated.
+    */
+    rotated : boolean;
+
+    /**
+    * boolean to flag is the frame was trimmed.
+    */
+    trimmed : boolean;
+
+    /**
+	 * ISpriteJSONSpriteFrame basic object of the source data
+	 */
+    spriteSourceSize : ISpriteJSONSpriteFrameData;
+
+    /**
+	 * ISpriteJSONSpriteFrame basic object of the source data
+	 */
+    sourceSize : ISpriteJSONSpriteSourceSize;
+}
+
+/**
+ * Defines the basic options interface of a JSON atlas.
+ */
+export interface ISpriteJSONAtlas{

```

Comment:
Do you mind moving them to individual files?

Reply:
which directory has an example with the interfaces split off into separate files so I can reference?

Reply:
Nevermind I figured it out.